tests/bluetooth/ctrl_isoal/src/main.c

```diff
@@ -20,1517 +20,7747 @@
+ * handle.
  */
-void test_unframed_padding_no_end(void)
+ZTEST(test_rx_basics, test_sink_invalid_ref)
 {
-	clear_sdu_buf();
-
-	/* Assumes 3 PDUs per SDU interval */
-
-	/* first padding in SDU interval */
-	pdu_ref = &testdata[0];
-	construct_pdu_unframed(PDU_BIS_LLID_START_CONTINUE, pdu_ref, 0 /* len = 0 => padding pdu */,
-			       payload_number++, 923749, 0);
-	err = isoal_rx_pdu_recombine(sink_hdl, &pdu_meta);
-	zassert_equal(err, ISOAL_STATUS_OK, "err=0x%02x", err);
-	/* Expecting 0 data to be written, with no error status */
-	zassert_equal(sink->sdu_production.sdu_written, 0, "written=%u",
-		      sink->sdu_production.sdu_written);
-	zassert_equal(sink->sdu_production.sdu_status, ISOAL_SDU_STATUS_VALID, "sdu_status=0x%x",
-		      sink->sdu_production.sdu_status);
-
-	/* Second padding in SDU interval */
-	pdu_ref = &testdata[0];
-	construct_pdu_unframed(PDU_BIS_LLID_START_CONTINUE, pdu_ref, 0 /* len = 0 => padding pdu */,
-			       payload_number++, 923750, 0);
-	err = isoal_rx_pdu_recombine(sink_hdl, &pdu_meta);
-	zassert_equal(err, ISOAL_STATUS_OK, "err=0x%02x", err);
-	/* Expecting 0 data to be written, with no error status */
-	zassert_equal(sink->sdu_production.sdu_written, 0, "written=%u",
-		      sink->sdu_production.sdu_written);
-	zassert_equal(sink->sdu_production.sdu_status, ISOAL_SDU_STATUS_VALID, "sdu_status=0x%x",
-		      sink->sdu_production.sdu_status);
-
-	/* Third and last padding in SDU interval, end not seen should result in error */
-	pdu_ref = &testdata[0];
-	construct_pdu_unframed(PDU_BIS_LLID_START_CONTINUE, pdu_ref, 0 /* len = 0 => padding */,
-			       payload_number++, 923751, 0);
-	err = isoal_rx_pdu_recombine(sink_hdl, &pdu_meta);
-	zassert_equal(err, ISOAL_STATUS_OK, "err=0x%02x", err);
-	/* Expecting 0 data to be written but with error status */
-	zassert_equal(sink->sdu_production.sdu_written, 0, "written=%u",
-		      sink->sdu_production.sdu_written);
-	zassert_equal(sink->sdu_production.sdu_status, ISOAL_SDU_STATUS_ERRORS, "sdu_status=0x%x",
-		      sink->sdu_production.sdu_status);
+	ztest_set_assert_valid(true);
+
+	isoal_get_sink_param_ref(99);
+
+	ztest_set_assert_valid(false);
 }
 
 /**
- * @brief   Padding PDU with error status received in the ISOAL_START state (beginning of SDU)
+ * Test Suite	:	RX basic test
+ *
+ * Test error return when receiving PDUs for a disabled sink.
  */
-void test_unframed_padding_error1(void)
+ZTEST(test_rx_basics, test_sink_disable)
 {
-	clear_sdu_buf();
+	struct rx_pdu_meta_buffer rx_pdu_meta_buf;
+	struct rx_sdu_frag_buffer rx_sdu_frag_buf;
+	struct isoal_sdu_buffer sdu_buffer;
+	isoal_sink_handle_t sink_hdl;
+	uint32_t stream_sync_delay;
+	uint32_t group_sync_delay;
+	isoal_sdu_len_t sdu_size;
+	uint8_t iso_interval_int;
+	uint64_t payload_number;
+	uint32_t pdu_timestamp;
+	uint32_t sdu_timestamp;
+	uint16_t testdata_indx;
+	uint16_t testdata_size;
+	uint32_t sdu_interval;
+	isoal_sdu_cnt_t seqn;
+	uint8_t testdata[23];
```

Comment:
Given that `23` is used multiple places, either use ARRAY_SIZE or put the test data length into a variable or `#define`

Reply:
It so happened that is ended up that way. It wasn't planned, but yes it could be defined.

Reply:
Still got time to correct it ;)